eox_nelp/notifications/tests/test_notify_course_due_date.py

```diff
@@ -0,0 +1,138 @@
+"""This file contains all the test for notifications/notify_course_due_date.py file.
+
+Classes:
+    NotifyUpcomingCourseDueDateTestCase: Test `notify_upcoming_course_due_date` method.
+"""
+from datetime import timedelta
+
+from django.test import TestCase
+from django.utils import timezone
+from mock import patch
+
+from eox_nelp.edxapp_wrapper.course_overviews import CourseOverview
+from eox_nelp.notifications import notify_course_due_date
+from eox_nelp.notifications.models import UpcomingCourseDueDate
+from eox_nelp.notifications.notify_course_due_date import notify_upcoming_course_due_date
+from eox_nelp.tests.utils import generate_list_mock_data
+
+
+class NotifyUpcomingCourseDueDateTestCase(TestCase):
+    """Test class for function `notify_upcoming_course_due_date`."""
+
+    def setUp(self):
+        """Set common conditions for test cases."""
+        self.course_name = "course-display-name"
+        self.subsection_name = "subsection-display-name"
+        self.recipients_list = ["vader@star.com", "palpatine@star.com"]
+        self.subsection_due_date = timezone.now() + timedelta(days=4)
+        self.enrollments_data = [
+            {"user": {"email": self.recipients_list[0]}},
+            {"user": {"email": self.recipients_list[1]}},
+        ]
+        self.subsection_xblock_data = [
+            {"display_name": self.subsection_name, "due": self.subsection_due_date},
+        ]
+        self.course_xblock_data = [
+            {
+                "display_name": self.course_name,
+            },
+        ]
+
+    @patch("eox_nelp.notifications.notify_course_due_date.send_email_multialternative")
+    @patch("eox_nelp.notifications.notify_course_due_date.CourseEmailTemplate")
+    @patch("eox_nelp.notifications.notify_course_due_date.get_xblock_course_from_usage_key_string")
+    @patch("eox_nelp.notifications.notify_course_due_date.get_xblock_from_usage_key_string")
+    @patch("eox_nelp.notifications.notify_course_due_date.CourseEnrollment")
```

Comment:
could you consider to move this to the setUp method, check this example https://github.com/eduNEXT/eox-nelp/blob/master/eox_nelp/signals/tests/test_tasks.py#L203

Reply:
[19b22cb](https://github.com/eduNEXT/eox-nelp/pull/44/commits/19b22cb8d382753eebe58111a49bb76ed96a3b69)